doc/mgmt/gnmi/save_on_set.md

```diff
@@ -0,0 +1,220 @@
+
+The new parameter will be: --with-save-on-set and when present it will configure a function pointer variable gnmi.SaveOnSet to point to a function that actually performs the save operation.
+
+```go
+var (
+  withSaveOnSet = flag.Bool("with-save-on-set", false, "Enables save-on-set.")
+)
+
+// ...
+
+if *withSaveOnSet {
+  gnmi.SaveOnSet = gnmi.SaveOnSetEnabled
+}
+```
+
+##### Call to Save ConfigDB
+A function to initial the backup located in the sonic-mgmt-framework.
+
+```go
+// SaveConfig initiates the operation of saving the current content of
+// the ConfigDB to a file that is then used to populate the database during the
+// startup.
+func SaveConfig() error {
+    r := HostQuery("cfg_mgmt.save", []string{})
+    if r.Err != nil{
+        return errors.New("internal SONiC Hostservice failure: " + r.Err.Error())
+    }
+    if len(r.Body) < 2 {
+        return errors.New(
+            "internal SONiC Hostservice failure: the response is too short.")
+    }
+    if _, ok := r.Body[0].(int32); !ok {
+        return errors.New(
+            "internal SONiC Hostservice failure: first element is not int32.")
+    }
+if_, ok := r.Body[1].(string); !ok {
+        return errors.New(
+            "internal SONiC Hostservice failure: second element is not string.")
+    }
+    if r.Body[0].(int32) != 0 {
+        return errors.New(r.Body[1].(string))
+    }
+    return nil
+}
+```
+
+##### gNMI.Set() Handler
+
+The RPC will execute the save-on-set function at the end of each call.
+
+```go
+// SaveOnSetEnabled saves configuration to a file
+func SaveOnSetEnabled() {
+  transformer.SaveStartupConfig()
+}
+// SaveOnSetDisabeld does nothing.
+func SaveOnSetDisabled() {}
+
+// SaveOnSet point to a function that is called to save changes of configuration
+// to a file. By default it points to an empty function - the configuration is not
+// saved to a file.
+var SaveOnSet = SaveOnSetDisabled
+
+func (s *Server) Set(ctx context.Context, req *gnmipb.SetRequest) (*gnmipb.SetResponse, error) {
+
+  // ...
+
+  SaveOnSet()
+  return resp, err
+}
+```
+
+### SAI API
+
+No change in SAI API.
+
+### Configuration and management
+
+#### gNMI
+
+Adds flag to enable feature in gNMI binary
+
+#### Manifest (if the feature is an Application Extension)
+
+Not an app extension.
+
+#### CLI/YANG model Enhancements
+
+No changes to CLI or YANG
+
+#### Config DB Enhancements
+
+Add an entry to Config DB for toggling the save-on-set feature.
+
+### Warmboot and Fastboot Design Impact
+
+No effect on warm/fast boot
+
+### Restrictions/Limitations
+
+### Testing Requirements/Design
+
+#### Unit Test cases
+
+- No behavior change if the feature is not enabled.
+- A gNMI.Set() call should generate a signal to save ConfigDB.
```

Comment:
Are we invoking "write memory" click call for every gNMI set operation when the save-on-set is enabled? not sure signalling is the word usage, please clarify.

Reply:
Reworded